### ISSUE TITLE
Custom builder: docker build of new image fails with "" is not a valid repository/tag

### ISSUE THREAD

**Issue by dolsem** [GoogleContainerTools/skaffold#3077]
I am trying to build a docker image with a custom build command (side note: I use custom builder instead of docker because docker builder doesn't seem to support `--ssh` option yet, which is available with Docker BuildKit).

### Expected behavior
Build succeeds.

### Actual behavior
Build fails with:
```
...
 => CACHED [14/14] RUN --mount=type=ssh npm install                                                  0.0s
 => exporting to image                                                                               0.1s
 => => exporting layers                                                                              0.0s
 => => writing image sha256:6b0cca15ca01ef16f26d159fcc9e1673ccc80537379d752e00dd234435e8fa6d         0.0s
 => => naming to docker.io/library/nest                                                              0.0s
FATA[0023] exiting dev mode because first build failed: build failed: build failed: building [nest]: Error parsing reference: "" is not a valid repository/tag: invalid reference format 
```

### Information

- Skaffold version: v0.40.0
- Operating system: Linux
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta16
kind: Config

build:
  local:
    push: false
  artifacts:
    - image: nest
      context: .
      sync:
        infer:
          - src/**/*.ts
          - .env.json
      custom:
        buildCommand: 'docker build -t nest -f Dockerfile.dev --ssh default .'

deploy:
  kubectl:
    manifests:
      - k8s/nest-node-port.yaml
      - k8s/nest-deployment.yaml
```
- Contents of Dockerfile.dev:
```dockerfile
# syntax=docker/dockerfile:experimental
FROM node:current-alpine

RUN apk add --no-cache \
  openssh-client git   \
  build-base           \
  bash python          \
  expect

RUN mkdir -m 0600 /root/.ssh/
RUN touch /root/.ssh/known_hosts
RUN ssh-keyscan gitlab.com >> /root/.ssh/known_hosts
RUN ssh-keyscan github.com >> /root/.ssh/known_hosts

WORKDIR /app

COPY package.json ./
COPY .env.json ./
COPY jest.* ./
COPY nodemon* ./
COPY tsconfig* ./
COPY scripts scripts
COPY src src

RUN --mount=type=ssh npm install

EXPOSE 3000
EXPOSE 3001

CMD ["npm", "run", "start:dev"]
```

### Steps to reproduce the behavior
Build a new docker image with custom build command and Docker BuildKit (I run `DOCKER_BUILDKIT=1 skaffold dev`). If there is a preexisting version of the same image (I tried removing BuildKit features and building with docker builder first), build succeeds with no error.

**Comment by dgageot** [comment]
Hi @dolsem, the issue come from the fact your custom build command always builds an image with tag `nest` instead of the tag provided by Skaffold.

The [doc](https://skaffold.dev/docs/how-tos/builders/#custom-build-script-run-locally) says that the custom build command should build an image with the tag passed as the `$IMAGES` env variable.

So you should edit your `skaffold.yaml` with:

```
custom:
  buildCommand: 'docker build -t $IMAGES -f Dockerfile.dev --ssh default .'
```

However, this won't work because of #3082. Until this issue is fixed with #3083, you have to create a `build.sh` script at the root of your project. See this example: https://github.com/GoogleContainerTools/skaffold/tree/master/examples/buildpacks

I also pushed a PR to improve the error message in such case: https://github.com/GoogleContainerTools/skaffold/pull/3085

**Comment by dolsem** [comment]
@dgageot oh I see, I missed that part in the docs. Good to know there's already a PR to fix the issue, thanks!
